traits/ctrait.py

```diff
@@ -109,6 +111,37 @@ def comparison_mode(self):
     def comparison_mode(self, value):
         ctraits.cTrait.comparison_mode.__set__(self, value)
 
+    @property
```

Comment:
I'm a little bit shocked and horrified that this works; my mental model of Python's name lookup rules just got broken ...

Reply:
To avoid reader confusion, let's follow Corran's suggestion and use `builtins.property` for this first decorator.

And as @rkern reminded me, we're in trouble if anyone then tries to add another property later on in the class definition. I'm wondering whether there's a better name than `property` that we could use here. Suggestions?

Reply:
How about using a trailing underscore `property_`? That way we also won't need to using `builtins.property`

Reply:
Picking a different name from `property` is a good idea anyway, since it'll help anyone external who's using this API (we _think_ that's no-one, but it's difficult to tell): breakage from a removed name is more obvious and easier to fix than breakage from changed behaviour for an existing name.

How about `property_getset`? It's not 100% accurate, since we sometimes have a validator in addition to a getter and setter (though that validator is often `None` in practice), but the name gives some idea of what it's for.

Reply:
Went with `property_fields` after a discussion on slack.